eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/immutable/AbstractImmutableBag.java

```diff
@@ -122,6 +122,17 @@
         return partitionMutableBag.toImmutable();
     }
 
+    /**
+     * @since 8.2
+     */
+    @Override
+    public <V> ImmutableBag<V> countBy(Function<? super T, ? extends V> function)
+    {
+        MutableBag<V> result = Bags.mutable.empty();
+        this.forEachWithOccurrences((each, count) -> result.addOccurrences(function.valueOf(each), count));
+        return result.toImmutable();
```

Comment:
No performance problem here. The implementation looks like it's the same as collect, though we don't implement collect on this class for performance reasons. I'd say actually delegate to collect, and consider making the default methods on the interfaces above regular abstract methods.